fastapi/routing.py

```diff
@@ -155,16 +154,20 @@ def get_request_handler(
 
     async def app(request: Request) -> Response:
         try:
-            body = None
+            body: Any = None
             if body_field:
                 if is_body_form:
                     body = await request.form()
                 else:
                     body_bytes = await request.body()
                     if body_bytes:
-                        body = await request.json()
-        except json.JSONDecodeError as e:
-            raise RequestValidationError([ErrorWrapper(e, ("body", e.pos))], body=e.doc)
+                        if (
+                            request.headers.get("Content-Type", "application/json")
```

Comment:
Is Content-Type not being there a concern? I believe RFC says it should always be filled.

Reply:
[RFC 7231](https://tools.ietf.org/html/rfc7231#section-3.1.1.5) suggests it is not strictly required, and that we should assume "application/octet-stream" if it's not present.